src/modelinstance.hpp

```diff
@@ -33,6 +33,7 @@
 
 #include "customloaderconfig.hpp"
 #include "customloaderinterface.hpp"
+#include "model_metric_reporter.hpp"
```

Comment:
Do all users of model instance need to know about metrics? If not then we should not include this header here. Then also you need to split this header to have separate impl file.

Reply:
Reporter contains inline functions to be used, those cannot be moved to cpp file